### ISSUE TITLE
Options currently not tested by RunTests

### ISSUE THREAD

**Issue by lalo** [VowpalWabbit/vowpal_wabbit#2867]
Generated with a script based on only the RunTests that are purely VW (skipping all bash tests). It also skips tests [195, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 258, 269]. So there might be some inaccuracies in this list. 

Elements with default values are on top, see below for ones without.

- [ ] OjaNewton, sketch_size, BUT has default value
- [ ] OjaNewton, epoch_size, BUT has default value
- [ ] OjaNewton, alpha, BUT has default value
- [ ] OjaNewton, learning_rate_cnt, BUT has default value
- [ ] active_cover, mellowness, BUT has default value
- [ ] active_cover, alpha, BUT has default value
- [ ] active_cover, beta_scale, BUT has default value
- [ ] active_cover, cover, BUT has default value
- [ ] baseline, baseline, BUT has default value
- [ ] baseline, global_only, BUT has default value
- [ ] baseline, check_enabled, BUT has default value
- [ ] bfgs, Conjugate Gradient options, conjugate_gradient, BUT has default value
- [ ] bfgs, LBFGS and Conjugate Gradient options, hessian_on, BUT has default value
- [ ] boosting, gamma, BUT has default value
- [ ] boosting, alg, BUT has default value
- [ ] cb_adf, no_predict, BUT has default value
- [ ] cb_adf, clip_p, BUT has default value
- [ ] cb_dro, cb_dro_alpha, BUT has default value
- [ ] cb_dro, cb_dro_tau, BUT has default value
- [ ] cb_dro, cb_dro_wmax, BUT has default value
- [ ] cb_explore, psi, BUT has default value
- [ ] cb_explore_adf_bag, greedify, BUT has default value
- [ ] cb_explore_adf_bag, first_only, BUT has default value
- [ ] cb_explore_adf_cover, psi, BUT has default value
- [ ] cb_explore_adf_cover, nounif, BUT has default value
- [ ] cb_explore_adf_cover, first_only, BUT has default value
- [ ] cb_explore_adf_greedy, first_only, BUT has default value
- [ ] cb_explore_adf_regcb, first_only, BUT has default value
- [ ] cb_explore_adf_rnd, rnd_alpha, BUT has default value
- [ ] cb_explore_adf_rnd, rnd_invlambda, BUT has default value
- [ ] cb_explore_adf_squarecb, gamma_exponent, BUT has default value
- [ ] cb_explore_adf_synthcover, synthcoverpsi, BUT has default value
- [ ] cb_explore_adf_synthcover, synthcoversize, BUT has default value
- [ ] cb_explore_pdf, epsilon, BUT has default value
- [ ] cb_explore_pdf, first_only, BUT has default value
- [ ] cbify, loss_report, BUT has default value
- [ ] cbify, loss_01_ratio, BUT has default value
- [ ] cbify, loss0, BUT has default value
- [ ] cbify, loss1, BUT has default value
- [ ] cbify_ldf, loss0, BUT has default value
- [ ] cbify_ldf, loss1, BUT has default value
- [ ] cs_active, baseline, BUT has default value
- [ ] cs_active, domination, BUT has default value
- [ ] cs_active, range_c, BUT has default value
- [ ] cs_active, max_labels, BUT has default value
- [ ] cs_active, min_labels, BUT has default value
- [ ] cs_active, cost_min, BUT has default value
- [ ] cs_active, csa_debug, BUT has default value
- [ ] ect, link, BUT has default value
- [ ] gd, sparse_l2, BUT has default value
- [ ] gd, l1_state, BUT has default value
- [ ] gd, l2_state, BUT has default value
- [ ] general, VW options, strict_parse, BUT has default value
- [ ] general, Weight options, random_weights, BUT has default value
- [ ] general, Weight options, normal_weights, BUT has default value
- [ ] general, Weight options, truncated_normal_weights, BUT has default value
- [ ] general, Parallelization options, unique_id, BUT has default value
- [ ] general, Parallelization options, total, BUT has default value
- [ ] general, Parallelization options, node, BUT has default value
- [ ] general, Parallelization options, span_server_port, BUT has default value
- [ ] general, Diagnostic options, version, BUT has default value
- [x] general, Diagnostic options, help, BUT has default value
- [ ] general, Feature options, permutations, BUT has default value
- [ ] general, Example options, holdout_period, BUT has default value
- [ ] general, Example options, early_terminate, BUT has default value
- [ ] general, Example options, sort_features, BUT has default value
- [ ] general, Example options, quantile_tau, BUT has default value
- [ ] general, Output model, save_per_pass, BUT has default value
- [x] general, Input options, daemon, BUT has default value
- [ ] general, Input options, foreground, BUT has default value
- [ ] general, Input options, compressed, BUT has default value
- [ ] general, Input options, no_daemon, BUT has default value
- [x] general, Input options, flatbuffer, BUT has default value
- [ ] ksvm, pool_greedy, BUT has default value
- [ ] ksvm, para_active, BUT has default value
- [ ] ksvm, pool_size, BUT has default value
- [ ] ksvm, subsample, BUT has default value
- [ ] ksvm, bandwidth, BUT has default value
- [ ] ksvm, degree, BUT has default value
- [ ] lda, lda_epsilon, BUT has default value
- [ ] lda, math-mode, BUT has default value
- [ ] lda, metrics, BUT has default value
- [ ] log_multi, no_progress, BUT has default value
- [ ] log_multi, swap_resistance, BUT has default value
- [ ] lrq, lrqdropout, BUT has default value
- [ ] marginal, update_before_learn, BUT has default value
- [ ] marginal, unweighted_marginals, BUT has default value
- [ ] memory_tree, top_K, BUT has default value
- [ ] memory_tree, oas, BUT has default value
- [ ] nn, inpass, BUT has default value
- [ ] nn, multitask, BUT has default value
- [ ] nn, dropout, BUT has default value
- [ ] nn, meanfield, BUT has default value
- [ ] noop, noop, BUT has default value
- [ ] plt, kary_tree, BUT has default value
- [ ] plt, threshold, BUT has default value
- [ ] pmf_to_pdf, first_only, BUT has default value
- [ ] print, print, BUT has default value
- [ ] rank, bfgs, BUT has default value
- [ ] rank, conjugate_gradient, BUT has default value
- [ ] recall_tree, bern_hyper, BUT has default value
- [ ] recall_tree, node_only, BUT has default value
- [ ] recall_tree, randomized_routing, BUT has default value
- [ ] replay_b, replay_b_count, BUT has default value
- [ ] replay_c, replay_c_count, BUT has default value
- [ ] replay_m, replay_m_count, BUT has default value
- [ ] search_task, search_beta, BUT has default value
- [ ] search_task, search_no_caching, BUT has default value
- [ ] search_task, search_xv, BUT has default value
- [ ] search_task, search_perturb_oracle, BUT has default value
- [ ] search_task, search_linear_ordering, BUT has default value
- [ ] svrg, svrg, BUT has default value
- [ ] svrg, stage_size, BUT has default value
- [ ] warm_cb, loss0, BUT has default value
- [ ] warm_cb, loss1, BUT has default value
- [ ] warm_cb, overwrite_label, BUT has default value
- [ ] OjaNewton, alpha_inverse
- [ ] OjaNewton, normalize
- [ ] OjaNewton, random_init
- [ ] baseline, lr_multiplier
- [ ] boosting, boosting
- [ ] cats_tree, link
- [ ] cb_explore_adf_bag, epsilon
- [ ] cb_explore_adf_first, epsilon
- [ ] cb_explore_adf_rnd, epsilon
- [ ] cb_explore_adf_softmax, epsilon
- [ ] csoaa_ldf, Cost Sensitive One Against All with Label Dependent Features, ldf_override
- [ ] explore_eval, multiplier
- [ ] general, Weight options, initial_weight
- [ ] general, Weight options, input_feature_regularizer
- [ ] general, Parallelization options, span_server
- [ ] general, Feature options, keep
- [ ] general, Feature options, feature_limit
- [ ] general, Feature options, spelling
- [ ] general, Feature options, q:
- [ ] general, Example options, initial_pass_length
- [ ] general, Example options, min_prediction
- [ ] general, Example options, max_prediction
- [ ] general, Output model, output_feature_regularizer_binary
- [ ] general, Output model, output_feature_regularizer_text
- [ ] general, Output model, id
- [ ] general, Output options, raw_predictions
- [x] general, Input options, port
- [x] general, Input options, num_children
- [ ] general, Input options, pid_file
- [ ] general, Input options, port_file
- [ ] new_mf, new_mf
- [ ] recall_tree, max_candidates
- [ ] recall_tree, max_depth
- [ ] replay_m, replay_m
- [ ] search_task, search_allowed_transitions
- [ ] search_task, search_subsample_time
- [ ] search_task, search_rollout_num_steps
- [ ] search_task, search_active_verify
- [ ] search_task, search_save_every_k_runs
- [ ] sendto, sendto

**Comment by lalo** [comment]
Reduction stacks exercised by RunTests:
```
gd->scorer, 36
gd->scorer->csoaa->search_task, 19
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->cb_sample->shared_feature_merger->ccb_explore_adf, 12
ftrl->scorer, 10
cbzo, 10
gd->scorer->csoaa_ldf->cb_adf->shared_feature_merger, 9
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_cover->shared_feature_merger, 8
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->shared_feature_merger->warm_cb, 7
gd->scorer->oaa, 6
gd->scorer->csoaa->cb->cb_explore->cbify, 6
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->shared_feature_merger, 6
gd->scorer->csoaa_ldf->shared_feature_merger, 5
gd->scorer->bootstrap, 5
gd->scorer->csoaa->cb, 5
gd->scorer->csoaa->cb->cb_explore, 5
gd->nn->scorer, 4
gd->stage_poly->scorer, 4
gd->scorer->csoaa_ldf->shared_feature_merger->search_task, 3
ksvm->scorer, 3
gd->active_cover->scorer->binary, 3
gd->scorer->multiworld_test, 3
gd->marginal->scorer, 3
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->cb_sample->shared_feature_merger->ccb_explore_adf->slates, 3
gd->scorer->csoaa->cb->cb_explore->ot, 3
gd->scorer->plt, 3
bfgs->scorer, 2
lda->scorer, 2
rank->scorer, 2
gd->scorer->csoaa, 2
gd->confidence->scorer, 2
gd->scorer->csoaa->cb->multiworld_test, 2
gd->scorer->recall_tree, 2
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_softmax->shared_feature_merger, 2
gd->scorer->classweight, 2
gd->scorer->cs_active, 2
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->shared_feature_merger->cbify, 2
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_regcb->shared_feature_merger->cbify, 2
gd->scorer->memory_tree, 2
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_squarecb->shared_feature_merger->cbify, 2
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_squarecb->shared_feature_merger->cbify_ldf, 2
ftrl->scorer->binary->cats_tree->get_pmf->pmf_to_pdf->cb_explore_pdf->cats_pdf->sample_pdf->cats->cbify, 2
ftrl->scorer->binary->cats_tree->get_pmf->pmf_to_pdf->cb_explore_pdf->cats_pdf->sample_pdf->cats, 2
ftrl->scorer->binary->cats_tree->get_pmf->pmf_to_pdf->cb_explore_pdf->cats_pdf, 2
gd->scorer->ect, 1
gd->active->scorer, 1
gd->scorer->top, 1
gd->scorer->binary, 1
gd->autolink->scorer, 1
gd->scorer->log_multi, 1
gd->lrq->scorer, 1
gd->scorer->multilabel_oaa, 1
gd->replay_b->scorer, 1
gd->scorer->csoaa->replay_c, 1
ftrl->nn->scorer->csoaa->search_task, 1
ftrl->scorer->audit_regressor, 1
gd->scorer->csoaa->replay_c->audit_regressor, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_bag->shared_feature_merger, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_first->shared_feature_merger, 1
OjaNewton->scorer, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->explore_eval->shared_feature_merger, 1
gd->lrqfa->scorer, 1
gd->scorer->audit_regressor, 1
gd->scorer->bootstrap->binary, 1
gd->scorer->bootstrap->oaa, 1
gd->scorer->oaa->classweight, 1
gd->scorer->recall_tree->classweight, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_regcb->shared_feature_merger->cbify_ldf, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_greedy->cb_sample->shared_feature_merger, 1
gd->scorer->csoaa_ldf->cb_adf->cb_dro->shared_feature_merger, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_rnd->shared_feature_merger, 1
ftrl->scorer->csoaa->cb->cb_explore->cbify, 1
ftrl->scorer->binary->cats_tree->cb_explore->cbify, 1
gd->scorer->csoaa_ldf->cb_adf->cb_explore_adf_synthcover->shared_feature_merger, 1
gd->scorer->interact, 1
```

**Comment by JohnLangford** [comment]
Coverage has now significantly increased.
